core/src/main/java/io/questdb/griffin/engine/QueryProgress.java

```diff
@@ -374,6 +404,67 @@ private static void appendLeakedReaderNames(ObjList<TableReader> leakedReaders,
         }
     }
 
+
+    private static String substituteBindVariablesIntoQueryText(@NotNull CharSequence sqlText, @NotNull SqlExecutionContext executionContext) {
+        StringSink sink = Misc.getThreadLocalSink();
+
+        int bindVarNumber;
+        for (int i = 0, n = sqlText.length(); i < n; i++) {
+            char c = sqlText.charAt(i);
+            int start;
+
+            switch (c) {
+                case '\'':
+                case '"':
+                    sink.put(c);
+                    for (int j = i + 1; j < n; j++) {
+                        c = sqlText.charAt(j);
+                        if (sqlText.charAt(j) == c) {
+                            break;
```

Comment:
we could reach the end of the text without finding the closing `'` or `"`.
i understand that this SQL is already parsed by the time we get here, so it must be valid.
would be good to add a comment to clarify this.